src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java

```diff
@@ -24,18 +28,19 @@
   }
 
   @Override
-  public void save(EmployeeDto employeeDto) {
+  public EmployeeDto save(EmployeeDto employeeDto) {
     if (employeeRepository.findByPhoneNumber(employeeDto.getPhoneNumber()).isPresent()) {
       ExceptionUtils.throwEntityAlreadyExistsException(
           "Employee already exists, choose a different phone Number");
-      return;
     }
 
     Employee employee = new Employee();
     BeanUtils.copyProperties(employeeDto, employee);
     employee.setUserName(employeeDto.getFirstName() + "." + employeeDto.getLastName());
 
```

Comment:
Don't make username on your own..let it come from client only..because 2 people can have same first and last name...but they can't have same username and employeeId...remove this...and let it come from client.

Reply:
okay nice comment